bls12_381/src/curves/g1.rs

```diff
@@ -40,6 +43,32 @@ impl SWModelParameters for Parameters {
     fn mul_by_a(_: &Self::BaseField) -> Self::BaseField {
         Self::BaseField::zero()
     }
+
+    fn is_in_correct_subgroup_assuming_on_curve(p: &GroupAffine<Parameters>) -> bool {
+        // Check that sigma(P) == -[X²]P
+        // if [X]P = P during the computation, return False
+        if p.is_zero() {
+            true
+        } else {
+            let sigma_p = sigma(p);
+            let mut x_times_p: GroupAffine<_> =
+                p.mul(BigInteger([Parameters0::X[0], 0, 0, 0])).into();
+            if Parameters0::X_IS_NEGATIVE {
+                x_times_p = -x_times_p;
+            }
+            if (-*p + x_times_p).is_zero() {
```

Comment:
```suggestion
            if (x_times_p - x).is_zero() {
```

Reply:
this is not working, and neither `x_times_p - *p`. The only way I found is to write `x_times_p + (-*p)`... Any suggestion?

Reply:
It was because x_times_p was `GroupAffine`. The following code should work:
```suggestion
            if x_times_p.add_mixed(-*p).is_zero() {
```